asserts/database.go

```diff
@@ -0,0 +1,110 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package asserts implements snappy assertions and a database
+// abstraction for managing and holding them.
+package asserts
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/crypto/openpgp/packet"
+
+	"github.com/ubuntu-core/snappy/helpers"
+)
+
+// DatabaseConfig for an assertion database.
+type DatabaseConfig struct {
+	// database backstore path
+	Path string
+}
+
+// Database holds assertions and can be used to sign or check
+// further assertions.
+type Database struct {
+	root string
+}
+
+const (
+	privateKeysLayoutVersion = "v0"
+	privateKeysRoot          = "private-keys-" + privateKeysLayoutVersion
+)
+
+// OpenDatabase opens the assertion database based on the configuration.
+func OpenDatabase(cfg *DatabaseConfig) (*Database, error) {
+	err := os.MkdirAll(cfg.Path, 0775)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create assert database root: %v", err)
+	}
+	info, err := os.Stat(cfg.Path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create assert database root: %v", err)
```

Comment:
Yeah, good point, I was wondering if it might be confusing if we ever get an error report about it to know what line caused it, but its really really minor :)